lib/src/rules/prefer_single_quotes.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = "Prefer single quotes where it won't require escape sequences";
+
+const _details = '''
+
+**DO** use single quotes where it wouldn't require additional escapes
```

Comment:
"where it" --> "for string literals where they"?

Also, if we're allowing the use of double quotes in other places we should describe the allowed uses.